components/eam/src/physics/rrtmgp/radiation.F90

```diff
@@ -976,26 +990,15 @@ subroutine radiation_init(state,pbuf)
 
       if (do_SPA_optics) then !initialize SPA
 
-         allocate(aerosol_optical_property_lw(pcols,pver,nlwbands,12,begchunk:endchunk))
-         allocate(aerosol_optical_property_sw(pcols,pver,nswbands,12,begchunk:endchunk))
-         do month = 1,12
-            call get_aerosol_optical_property_from_file(month,'AER_TAU_LW','lwband',nlwbands,aerosol_optical_property_lw(:,:,:,month,:))
-         end do
-         call pbuf_set_field(pbuf,aer_tau_bnd_lw_mon_idx,aerosol_optical_property_lw)
-         do month = 1,12 
-            call get_aerosol_optical_property_from_file(month,'AER_TAU_SW','swband',nswbands,aerosol_optical_property_sw(:,:,:,month,:))
+         do band_index = 0,nswbands
+            aer_tau_bnd_sw_idx(band_index) = pbuf_get_index(aer_tau_sw_names(band_index))
+            aer_ssa_bnd_sw_idx(band_index) = pbuf_get_index(aer_ssa_sw_names(band_index))  
+            aer_asm_bnd_sw_idx(band_index) = pbuf_get_index(aer_asm_sw_names(band_index))
```

Comment:
Stuffing the 4d vector needed by rad with the pointers to the 3d vectors available from tracer_data is really slick. We should make sure it doesn't cause big performance issues due to weird striding. I don't think it will since we're pulling from contiguous assignments to pbuf, but just in case...